File: 2015/s2.cpp
// code by savir singh

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main() {
    ll j, a;
    cin >> j >> a;
    vector<char> jerseys(j);
    ll counter=0;
    for (int i=0; i<j; i++) {
        cin >> jerseys[i];
    }
    for (int i=0; i<a; i++) {
        char c;
        ll d;
        cin >> c >> d;
        d--;
        if (jerseys[d]==c || int(jerseys[d])<int(c)) {
            counter++;
            jerseys[d]='Z';
        }
    }
    cout << counter;
}

File: 2015/s2.py
# code by savir singh

# returns TLE cuz Python is slowwwww
# code works fully though
# i might make this algo more efficient later
# check C++ code - it works
# 12/15 points

j = int(input())
a = int(input())
jerseys = []
athletes = []
requests = 0
orders = ['S', 'M', 'L']
taken = []

for i in range(j):
    jer = input()
    jerseys.append(jer)

for c in range(a):
    ath = input().split(" ")
    athletes.append(ath)

for k in range(len(athletes)):
    thejersey = jerseys[int(athletes[k][1])-1]
    correctjersey = athletes[k][0]
    if thejersey == correctjersey or orders.index(thejersey) > orders.index(correctjersey):
        if int(athletes[k][1]) not in taken:
            requests += 1
        taken.append(int(athletes[k][1]))

print(requests)
